clients/python-wrapper/lakefs/branch.py

```diff
@@ -18,7 +18,16 @@ class Branch(Reference):
     Class representing a branch in lakeFS.
     """
 
-    def create(self, source_reference_id: str, exist_ok: bool = False) -> Branch:
+    def _get_commit(self):
+        """
+        For branches override the default _get_commit method to ensure we always fetch the latest head
+        """
+        with api_exception_handler():
+            commit = self._client.sdk_client.commits_api.get_commit(self._repo_id, self._id)
+            self._commit = Commit(**commit.__dict__)
```

Comment:
Why are we caching the current HEAD? The point of a branch is to allow mutation and for HEAD to change.. Perhaps I'm missing something.

Reply:
```suggestion
            self._commit = Commit(**commit.dict())
```

Pydantic gives you a public method for this

Reply:
Prefer to defer it to another PR where I'll change all occurrences of this code fragment

Reply:
Done

Reply:
> Why are we caching the current HEAD? The point of a branch is to allow mutation and for HEAD to change.. Perhaps I'm missing something.

We're not caching the branch head. the self._commit is inherited from the Reference class. For commits and tags which are immutable we don't to fetch the commit information each time so we do it only once. In branch we want to override this behavior by explicitly getting the head each time

Reply:
Cleaned up the code